### PR TITLE
Automatic and persisted theme choice 

### PR DESCRIPTION
**Pull request by Nomandes**
# Automatic and persisted theme choice 

The theme is now chosen by the system wide setting with the help of the media query "prefers-color-scheme". Additionally the manually chosen theme is persisted in local storage and overwrites the media query choice. The implementation is prepared to handle more themes in the future

### CODE REVIEW

File: packages/frontend/src/common/context/ThemeContext.tsx [hunk 1]
```diff
@@ -1,115 +1,41 @@
...
-  function setLightTheme() {
-    setCurrentTheme(lightTheme);
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+  const [themeChoice, setThemeChoice] = useState<RetroPaletteMode | undefined>(undefined);
+
+  const currentTheme: Theme = useMemo(() => {
+    if (themeChoice === undefined || themes[themeChoice] === undefined) {
+      return prefersDarkMode ? (themes["dark"] as Theme) : (themes["light"] as Theme);
+    }
+    return themes[themeChoice] as Theme;
```
Comment (Nomandes) [inline packages/frontend/src/common/context/ThemeContext.tsx:24]: Had to define the types to ensure typescript that the Record wouldn´t be undefined

File: packages/frontend/mui.types.d.ts [hunk 2]
```diff
@@ -32,3 +32,5 @@ declare module "@mui/material/Typography" {
     label: true;
   }
 }
+
+export type RetroPaletteMode = "highContrast" | PaletteMode;
```
Comment (Nomandes) [inline packages/frontend/mui.types.d.ts:36]: Added a mode which do not exist yet as preparation for more themes

File: packages/frontend/src/common/context/ThemeContext.tsx [hunk 3]
```diff
@@ -1,115 +1,41 @@
...
+  }, [prefersDarkMode, themeChoice]);
+
+  useLocalStorage(() => {
+    setThemeChoice(LocalStorage.getThemePreference());
+  });
```
Comment (NearW) [inline packages/frontend/src/common/context/ThemeContext.tsx:29]:
```suggestion
export function ThemeContextProvider(props: PropsWithChildren) {
  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
  const defaultTheme: RetroPaletteMode = prefersDarkMode ? "dark" : "light";
  const [themeChoice, setThemeChoice] = useState<RetroPaletteMode>(defaultTheme);

  const currentTheme = themes[themeChoice];

  useLocalStorage(() => {
    setThemeChoice(LocalStorage.getThemePreference() ?? defaultTheme);
  });
```

File: packages/frontend/mui.types.d.ts (hunk 2 above)
Comment (NearW) [reply packages/frontend/mui.types.d.ts:36]:
```suggestion
export type RetroPaletteMode = "light" | "dark";
```

using `| PaletteMode` results in an `any` type for some reason. Let's omit this and just define the enum ourselves.

Comment (Nomandes) [reply packages/frontend/mui.types.d.ts]: Type resulted in any due to a missing import statement for PaletteMode. In current implmentation the type does not exist anymore

File: packages/frontend/src/common/components/Footer.tsx [hunk 4]
```diff
@@ -35,12 +35,12 @@ export default function Footer() {
           fontSize="0.8rem"
         >
           {links.map(({ href, label }, index) => (
-            <>
+            <React.Fragment key={index + href}>
               <Link href={href} underline="hover" target="_blank" rel="noopener" key={index + href}>
```
Comment (NearW) [inline packages/frontend/src/common/components/Footer.tsx:39]: This key can be removed

**Review by NearW — APPROVED** [review]
Otherwise LGTM

**Pull request merged**